base/db/R/query.pft.R

```diff
@@ -45,6 +47,57 @@ query.pft_species <- function(pft, modeltype, con) {
 }
 #==================================================================================================#
 
+##' Select cultivars associated with a PFT
+##'
+##' Given a PFT name and optionally a modeltype, finds its pft_id and
+##' returns the cultivars associated with it.
+##'
+##' @details A PFT is allowed to have associated species or associated
+##' cultivars, but not both. If this function returns no results, try checking
+##' your PFT with \code{\link{query.pft_species}} instead.
+##' Note that the cultivars associated with one PFT *are* allowed to come from
+##' multiple species, if desired.
+##'
+##' @inheritParams query.pft_species
+##' @return tibble containing names and ids for each cultivar
+##'   and the species it comes from
+##' @export
+query.pft_cultivars <- function(pft, modeltype = NULL, con) {
+
+  pft_tbl <- (dplyr::tbl(con, "pfts")
+    %>% dplyr::filter(name == pft, pft_type == "cultivar"))
+
+  if (!is.null(modeltype)) {
+    pft_tbl <- (pft_tbl
+      %>% dplyr::inner_join(
+        dplyr::tbl(con, "modeltypes"),
+        by = c("modeltype_id" = "id"),
+        suffix = c("", ".mt"))
+      %>% dplyr::filter(name.mt == modeltype))
+  }
+
+  (pft_tbl
```

Comment:
shouldn't this be captured as a new object and explicitly returned?

Reply:
Not if we're using Adv-R style: ["Only use return() for early returns."](http://r-pkgs.had.co.nz/style.html#style-fun)

Reply:
OK. thanks for checking